docs/components/concepts/job-workers.md

```diff
@@ -123,23 +123,26 @@ To help visualize the process in general, here is a sequence diagram which shows
+To avoid workers overloaded with too many jobs, e.g. running out of memory:
 
-Essentially, as jobs are pushed downstream from the broker to the client, they're first buffered in the gateway where the direct client connection reside. The gateway only sends as much data as the client can consume over a specific connection. If it notices its send buffers fill up, it will mark a client as `not-ready`. This can happen, for example, if the client's receive method is blocked/suspended.
+- The workers rely on the [built-in gRPC flow control mechanism](https://grpc.io/docs/guides/flow-control/).
+- Or, if lacking for your language of choice, [the built-in HTTP/2 stream flow control](https://httpwg.org/specs/rfc7540.html#FlowControl), e.g. Golang implementation of gRPC.
 
-If a client is not ready to receive a job, the gateway will instead try to re-route the job to another, logically equivalent worker. If this fails (e.g. all workers connected to a specific gateway are not ready), the job is returned to the broker. There, it may be retried to another gateway, if and only if it has a logically equivalent worker.
+Essentially, as jobs are pushed downstream from the broker to the client, they're first buffered in the gateway where the direct client connection resides. The gateway only sends as much data as the client can consume over a specific connection. If it notices its send buffers fill up, it marks a client as `not-ready`. This can happen, for example, if the client's receive method is blocked/suspended.
+
+If a client is not ready to receive a job, the gateway instead tries to re-route the job to another logically equivalent worker. If this fails (e.g. all workers connected to a specific gateway are not ready), the job is returned to the broker. There, it may be retried to another gateway, if and only if it has a logically equivalent worker.
 
 #### Implementing backpressure
 
-If you're using the raw `StreamActivatedJobs` RPC, or want to add support for this to your client of choice, the criteria to apply back pressure is to stall the underlying HTTP/2 transport. To do so, you may need to block the thread in which the gRPC stream is running (e.g. Java), or suspend the coroutine (e.g. Kotlin, Go). Once the transport stops receiving, this will cause the gateway's send buffers to fill up, and effectively apply back pressure.
+If you're using the raw `StreamActivatedJobs` RPC, or want to add support for this to your client of choice, the criteria to apply backpressure is to stall the underlying HTTP/2 transport. To do so, you may need to block the thread in which the gRPC stream is running (e.g. Java), or suspend the coroutine (e.g. Kotlin, Go). Once the transport stops receiving, this causes the gateway's send buffers to fill up, and effectively apply backpressure.
```

Comment:
Out of curiosity @christinaausley since I run into this issue quite often, especially with Google docs or Grammarly 🤔 I think I use often backpressure or the other one (not sure 😆 ) and it sometimes use the other one 🤷🏼

Reply:
Backpressure works just fine. I'll be sure to add this to the glossary and announce in the next WCoE release 😄